apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -1388,13 +1405,15 @@ def delete_port_postcommit(self, context):
             self._delete_path_if_last(context)
             self._release_dynamic_segment(context)
         if port.get('device_owner') == n_constants.DEVICE_OWNER_ROUTER_GW:
-            if self._is_nat_enabled_on_ext_net(network):
-                self._delete_shadow_ext_net_for_nat(context, port, network)
-            else:
-                self._delete_gw_port_nat_disabled(context)
+            if self.fabric_l3:
```

Comment:
wouldn't his be a little more natural to write as:

if self.fabric_l3:
   if gateway-port:
          ....
   elif interface-port:
          ...